org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/AbstractNullAnnotationTest.java

```diff
@@ -81,16 +82,29 @@ protected void setUp() throws Exception {
 		setUpAnnotationLib();
 	}
 
-	/**
-	 * @deprecated indirectly uses deprecated class PackageAdmin
-	 */
 	protected void setUpAnnotationLib() throws IOException {
 		if (this.LIBS == null) {
 			String[] defaultLibs = getDefaultClassPaths();
 			int len = defaultLibs.length;
 			this.LIBS = new String[len+1];
 			System.arraycopy(defaultLibs, 0, this.LIBS, 0, len);
 			this.LIBS[len] = getAnnotationLibPath();
+
+			Bundle[] bundles = FrameworkUtil.getBundle(this.getClass()).getBundleContext().getBundles();
+
+			Bundle bundle= null;
+			for (int i = 0; i < bundles.length; i++) {
+				if (bundles[i].getSymbolicName().equals("org.eclipse.jdt.annotation")) {
+					bundle=bundles[i];
+					break;
+				}
+			}
+
+			File bundleFile = FileLocator.getBundleFileLocation(bundle).get();
```

Comment:
Couldn't this be simplified to the following suggestion?
```suggestion
			File bundleFile = Optional.of(Platform.getBundle("org.eclipse.jdt.annotation"))
					.flatMap(FileLocator::getBundleFileLocation).orElseThrow();
```
And then it's basically the same as in the called `getAnnotationLibPath()`.
Or isn't it sufficient because `Platform.getBundle()` discards bundles that are in `INSTALLED`/`UNINSTALLED` mode?
And even then I would make this a bit more compact using a stream pipeline:
```suggestion
			Bundle[] bundles = FrameworkUtil.getBundle(this.getClass()).getBundleContext().getBundles();
			File bundleFile = Arrays.stream(bundles)
					.filter(b -> "org.eclipse.jdt.annotation".equals(b.getSymbolicName())) //
					.findFirst().flatMap(FileLocator::getBundleFileLocation).orElseThrow();
```